### PR TITLE
Exchange state selection page

### PR DESCRIPTION
**Pull request by schnogz**
## Description
Adds state selection page for US residents trying to use Shapeshift

## Change Type
Feature

## Testing Steps
Create new wallet using VPN mocking US IP, go to exchange tab and see the state selection page, select whitelisted state and press continue. User should now be able to use exchange.  

Log out of wallet and back in and the state selection box should not reappear as data was persisted in kvStore.

## PR Creator Checklist
- [x] Code compiles correctly
- [x] No lint issues exist (verified via `yarn lint`)
- [x] All unit tests pass (verified via `yarn test`)
- [x] Updated `README.md` and other documentation (if necessary)

## PR Reviewer Checklist
- [ ] Change validated and application spot checked
- [ ] Code styles and best practices met
- [ ] Code is readable and commented when necessary

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/scenes/Exchange/Shapeshift/selectors.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { lift } from 'ramda'
+
+export const getData = state => {
+  const userCountryR = selectors.core.settings.getCountryCode(state)
+  const userStateR = selectors.core.kvStore.shapeShift.getUsState(state)
+  const exchangeStep = selectors.components.exchange.getStep(state)
+
+  const transform = (country, usState) => {
+    // determine if user is in the US and has not registered their state with Shapeshift
+    const isRegistered = country === 'US' && usState && usState.Code
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/Exchange/Shapeshift/selectors.js]: Only user's in the US have to select a state. If a user from outside US who has been using shapeshift accesses Exchange, or someone outside US accesses the page for the first time isRegistered will be false and step will be 0. I think the check for isRegistered should be `country !== 'US' || usState`

File: packages/blockchain-wallet-v4/src/redux/kvStore/shapeShift/reducers.js [hunk 2]
```diff
@@ -22,6 +23,9 @@ export default (state = INITIAL_STATE, action) => {
     case AT.FETCH_METADATA_SHAPESHIFT_FAILURE: {
       return Remote.Failure(payload)
     }
+    case AT.ADD_STATE_METADATA_SHAPESHIFT: {
+      return set(compose(mapped, value, lensPath(['USAState', 'Code'])), payload.usState, state)
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/kvStore/shapeShift/reducers.js]: `USAState` must be saved as a `string` so iOS and Android can correctly parse. Currently it's saved as {Code: 'State'}

**Comment by plondon** [comment]
Also small request but can you add a border around the state selection component so it matches v3 styling?

File: packages/blockchain-wallet-v4-frontend/src/scenes/Exchange/Shapeshift/selectors.js (hunk 1 above)
Comment (schnogz) [reply packages/blockchain-wallet-v4-frontend/src/scenes/Exchange/Shapeshift/selectors.js]: fixed

**Comment by schnogz** [comment]
I've addressed all PR comments and tested.  Should be good to go!

**Pull request merged**